.github/workflows/wheels-test.yml

```diff
@@ -107,9 +112,15 @@ jobs:
           TEST_MATRIX=$(yq -n 'env(MATRICES) | .[strenv(BUILD_TYPE)]')
           export TEST_MATRIX
 
+          echo ${{ inputs.pure_wheel }}
+          if [[ ${{ inputs.pure_wheel }} == "true" ]]; then
+              pure_wheel_filter='[.[] | select(.arch == "amd64" and .image == "ubuntu18.04" and ."test-type" == "unit" and .python == "3.10")]'
+          else
+              pure_wheel_filter='.'
+          fi
```

Comment:
We decided to hard-code the filter for pure python wheels so that the things it is selecting are close to the point of definition. Otherwise when you update the matrix here you have to go round the houses everywhere and match things up.